src/main/java/org/jboss/modules/ModuleLoader.java

```diff
@@ -574,13 +575,33 @@ protected final Module findLoadedModuleLocal(String name) {
      * @param module the module to unload
      * @throws SecurityException if an attempt is made to unload a module which does not belong to this module loader
      * @throws SecurityException if the module was not defined by this module loader
+     * @deprecated use {@link #unloadModuleLocal(String,Module)} instead
      */
+    @Deprecated
     protected final void unloadModuleLocal(Module module) throws SecurityException {
+        unloadModuleLocal(null, module);
```

Comment:
The contract of the two-argument version seems useful: "Unload the module name xxx if it still refers to actual module yyy".  This might be useful to avoid problems in racy situations (like concurrent deploy and undeploy), but such a method should also return a boolean to indicate that the change was successfully made.